src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -59,7 +70,17 @@ const createReleaseServiceInstalls = async (
 		options: {
 			$select: ['device', 'installs__service'],
 			$filter: {
-				device: { $in: deviceIds },
+				// Pass the device filters instead of IDs, since a using $in errors with `code: '08P01'` for more than 66k IDs.
+				device: Array.isArray(deviceFilterOrIds)
+					? { $in: deviceFilterOrIds }
+					: {
+							$any: {
+								$alias: 'd',
+								$expr: {
+									d: deviceFilterOrIds,
+								},
+							},
+					  },
```

Comment:
Arguably, doing a $in might be improving the performance of this query for smaller fleets, but on the other hand I ended up preferring a uniform approach depending on the provided parameter type, rather than creating a fleet size based edge case.